subsys/nrf_security/src/drivers/cracen/cracenpsa/src/key_management.c

```diff
@@ -429,6 +426,89 @@ static psa_status_t import_rsa_key(const psa_key_attributes_t *attributes, const
 	return PSA_SUCCESS;
 }
 
+static psa_status_t import_spake2p_key(const psa_key_attributes_t *attributes, const uint8_t *data,
+				       size_t data_length, uint8_t *key_buffer,
+				       size_t key_buffer_size, size_t *key_buffer_length,
+				       size_t *key_bits)
+{
+	size_t bits = psa_get_key_bits(attributes);
+	psa_key_type_t type = psa_get_key_type(attributes);
+
+	switch (type) {
+	case PSA_KEY_TYPE_SPAKE2P_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1):
+		if (data_length != 64) {
+			return PSA_ERROR_NOT_SUPPORTED;
+		}
+		if (bits != 0 && (bits != 256)) {
+			return PSA_ERROR_INVALID_ARGUMENT;
+		}
+		if (constant_memcmp_is_zero(data, 32)) {
+			return PSA_ERROR_INVALID_ARGUMENT;
+		}
+		if (constant_memcmp_is_zero(data + 32, 32)) {
+			return PSA_ERROR_INVALID_ARGUMENT;
+		}
```

Comment:
I assume we do constant_memcmp to avoid timing analysis.

Do we care that attackers can know which of the two 32bit fields in the data are zero?

Reply:
No, we can compare the "normal" way here